src/finch/interface/lazy.py

```diff
@@ -1,79 +1,124 @@
-import operator
 import builtins
+import itertools
+import operator
 from collections.abc import Callable
 from dataclasses import dataclass
-from typing import Any, Tuple, Iterable
-from itertools import accumulate
-from numpy.core.numeric import normalize_axis_index, normalize_axis_tuple
-from ..algebra import *
-
-from ..finch_logic import *
+from typing import Any
+
+from numpy.core.numeric import normalize_axis_tuple
+
+from ..algebra import (
+    element_type,
+    fill_value,
+    fixpoint_type,
+    init_value,
+    register_property,
+    return_type,
+)
+from ..finch_logic import (
+    Aggregate,
+    Alias,
+    Field,
+    Immediate,
+    LogicNode,
+    MapJoin,
+    Plan,
+    Produces,
+    Query,
+    Relabel,
+    Reorder,
+    Subquery,
+    Table,
+)
+from ..finch_logic.interpreter import FinchLogicInterpreter
 from ..symbolic import gensym
 
+__all__ = [
+    "LazyTensor",
+    "lazy",
+    "get_default_scheduler",
+    "compute",
+    "permute_dims",
+    "identify",
+    "expand_dims",
+    "squeeze",
+    "reduce",
+    "elementwise",
+    "prod",
+    "multiply",
+]
+
+
 @dataclass
 class LazyTensor:
     data: LogicNode
-    shape: Tuple
-    fill_value: Any
+    shape: tuple[int, ...]
+    fill_value: Immediate
```

Comment:
The fill value is inconvenient as an Immediate, then we need to call .val on it every time we want to use it.